### PR TITLE
EOS-21253 dtm0: implement MD path instrumentation with traces

### PR DESCRIPTION
**Pull request by sergey-shilov**
This patch introduces the instrumentation based on m0traces
for whole meta-data stack from client side to CAS fom.

Signed-off-by: Sergey Shilov <sergey.shilov@seagate.com>

### CODE REVIEW

File: addb2/dump.c [hunk 1]
```diff
@@ -1131,8 +1131,8 @@ struct m0_addb2__id_intrp ids[] = {
 
 	{ M0_AVI_RPC_LOCK,        "rpc-machine-lock", { &ptr } },
 	{ M0_AVI_RPC_REPLIED,     "rpc-replied",      { &ptr, &rpcop } },
-	{ M0_AVI_RPC_OUT_PHASE,   "rpc-out-phase",    { &rpc_out, SKIP2 } },
-	{ M0_AVI_RPC_IN_PHASE,    "rpc-in-phase",    { &rpc_in, SKIP2 } },
+	{ M0_AVI_RPC_OUT_STATE,   "rpc-out-state",    { &rpc_out, SKIP2 } },
+	{ M0_AVI_RPC_IN_STATE,    "rpc-in-state",    { &rpc_in, SKIP2 } },
 	{ M0_AVI_RPC_ITEM_ID_ASSIGN, "rpc-item-id-assign",
```
Comment (just-now) [inline addb2/dump.c:1136]: What the reason for such rename?

File: fop/fom.c [hunk 2]
```diff
@@ -1760,6 +1786,12 @@ M0_INTERNAL struct m0_reqh *m0_fom2reqh(const struct m0_fom *fom)
 	return fom->fo_service->rs_reqh;
 }
 
+M0_INTERNAL void m0_fom_phase_trace_enable(struct m0_fom *fom)
+{
+	fom->fo_sm_trace_off = false;
+}
+
+
```
Comment (just-now) [inline fop/fom.c:1794]: remove 1794.

File: addb2/dump.c (hunk 1 above)
Comment (sergey-shilov) [reply addb2/dump.c:1136]: The reason was just to make it aligned with names of states of other entities like:
- cas-req-state
- dix-req-state
- op-state
- ioo-req-state
- cob-req-state
- stio-req-state

Also per my understanding word "state" has more precise meaning here then "phase".

File: fop/fom.c (hunk 2 above)
Comment (sergey-shilov) [reply fop/fom.c:1794]: Done.

**Review by just-now — APPROVED** [review]

**Pull request merged**